### PR TITLE
storage: add bucket lifecycle management

### PR DESCRIPTION
**Pull request by jkwlui**

### CODE REVIEW

File: storage/api/Storage/Storage.cs [hunk 1]
```diff
@@ -1243,6 +1245,46 @@ private void GetUniformBucketLevelAccess(string bucketName)
...
+            var bucket = storage.GetBucket(bucketName);
+            bucket.Lifecycle = new Bucket.LifecycleData
+            {
+                Rule = new List<Bucket.LifecycleData.RuleData>
+                {
+                    new Bucket.LifecycleData.RuleData
+                    {
+                        Condition = new Bucket.LifecycleData.RuleData.ConditionData{
+                            Age = 100,
+                        },
```
Comment (amanda-tarafa) [inline storage/api/Storage/Storage.cs:1261]: nit: either 
```suggestion
                        Condition = new Bucket.LifecycleData.RuleData.ConditionData { Age = 100, },
```

or
```suggestion
                        Condition = new Bucket.LifecycleData.RuleData.ConditionData
                        {
                            Age = 100,
                        },
```

**Comment by frankyn** [comment]
@amanda-tarafa not urgent. @laljikanjareeya could you include these samples into your on-going PR #973?

**Comment by laljikanjareeya** [comment]
> @amanda-tarafa not urgent. @laljikanjareeya could you include these samples into your on-going PR #973?

@frankyn  Sure.

**Comment by amanda-tarafa** [comment]
> > @amanda-tarafa not urgent. @laljikanjareeya could you include these samples into your on-going PR #973?
> 
> @frankyn Sure.

Thanks both!

**Pull request closed without merging**